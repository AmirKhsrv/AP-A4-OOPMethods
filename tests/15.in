best_reviewer
show_author_info 461
show_sorted_shelf 102 read Romance
recommend_first_approach 129
recommend_first_approach 91
best_reviewer
recommend_first_approach 75
credit 24
recommend_second_approach 74
best_reviewer
best_book
recommend_first_approach 2
recommend_second_approach 53
recommend_second_approach 35
best_book
