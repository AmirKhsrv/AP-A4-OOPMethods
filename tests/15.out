id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
id: 461
Name: Amanda Downum
Year of Birth: 1970
Place of Birth: Madagascar
Member Since: March 13 2014
Genres: Science Fiction
Books:
id: 74
Title: Fated
Genre: Romance
Author: S.H. Kolee
***
id: 123
Title: Lead
Genre: Romance
Author: Kylie Scott
***
id: 44
Title: The V Girl
Genre: Romance
Author: Mya Robarts
***
id: 169
Title: A Leader's Gift: How to Earn the Right to Be Followed
Genre: Business
Author: Barry Banther
***
id: 185
Title: Broken World
Genre: Horror
Author: Kate L. Mary
***
id: 146
Title: Forever Fae
Genre: Fantasy
Author: L.P. Dover
***
id: 288
Title: Wander Dust
Genre: Fantasy
Author: Michelle Warren
***
id: 357
Title: Dragon Rider
Genre: Fantasy
Author: Cornelia Funke
Reviews:
id: 237 Rating: 1 Likes: 0 Date: March 19 2019
id: 807 Rating: 0 Likes: 0 Date: August 25 2016
id: 1205 Rating: 5 Likes: 0 Date: March 14 2018
id: 2621 Rating: 4 Likes: 0 Date: July 14 2017
id: 2748 Rating: 0 Likes: 0 Date: March 9 2016
id: 3126 Rating: 1 Likes: 0 Date: September 6 2017
id: 3394 Rating: 2 Likes: 0 Date: February 21 2016
id: 3594 Rating: 5 Likes: 0 Date: July 26 2018
id: 3689 Rating: 2 Likes: 0 Date: September 19 2016
id: 3908 Rating: 4 Likes: 0 Date: August 29 2019
id: 4156 Rating: 2 Likes: 10 Date: November 19 2017
id: 4336 Rating: 0 Likes: 52 Date: March 13 2019
id: 4661 Rating: 3 Likes: 0 Date: May 4 2015
id: 5057 Rating: 4 Likes: 0 Date: August 28 2019
id: 6502 Rating: 4 Likes: 5 Date: August 1 2016
id: 6702 Rating: 1 Likes: 0 Date: January 22 2017
id: 6756 Rating: 5 Likes: 6 Date: January 24 2018
id: 6811 Rating: 2 Likes: 0 Date: September 1 2019
id: 6824 Rating: 4 Likes: 10 Date: July 10 2016
id: 8132 Rating: 4 Likes: 0 Date: August 14 2015
id: 8520 Rating: 0 Likes: 0 Date: October 2 2017
id: 8965 Rating: 3 Likes: 0 Date: May 17 2018
id: 8986 Rating: 1 Likes: 0 Date: November 18 2016
id: 9012 Rating: 3 Likes: 0 Date: May 16 2019
id: 9278 Rating: 2 Likes: 146 Date: February 29 2013
id: 9309 Rating: 4 Likes: 0 Date: January 10 2017
id: 9758 Rating: 4 Likes: 0 Date: April 13 2018
id: 10411 Rating: 5 Likes: 0 Date: January 13 2019
id: 11475 Rating: 3 Likes: 0 Date: November 16 2019
id: 12798 Rating: 4 Likes: 0 Date: February 24 2019
id: 12866 Rating: 5 Likes: 0 Date: May 26 2014
id: 13003 Rating: 2 Likes: 0 Date: September 20 2015
id: 13316 Rating: 4 Likes: 0 Date: August 15 2016
id: 13330 Rating: 0 Likes: 1 Date: June 18 2015
id: 13589 Rating: 2 Likes: 0 Date: November 11 2015
id: 14238 Rating: 4 Likes: 4 Date: January 16 2012
id: 14588 Rating: 0 Likes: 0 Date: July 2 2016
id: 14619 Rating: 0 Likes: 0 Date: December 9 2019
id: 14678 Rating: 3 Likes: 1 Date: January 18 2019
id: 14985 Rating: 0 Likes: 0 Date: October 14 2019
id: 15249 Rating: 3 Likes: 0 Date: December 29 2019
id: 15753 Rating: 0 Likes: 1 Date: April 22 2018
id: 16044 Rating: 1 Likes: 1 Date: March 17 2017
id: 16148 Rating: 4 Likes: 0 Date: July 16 2019
id: 16478 Rating: 5 Likes: 0 Date: February 26 2014
id: 16649 Rating: 2 Likes: 0 Date: October 8 2016
id: 16992 Rating: 5 Likes: 0 Date: August 1 2016
id: 17039 Rating: 4 Likes: 0 Date: March 6 2019
id: 17101 Rating: 5 Likes: 0 Date: January 14 2016
id: 17406 Rating: 5 Likes: 0 Date: March 21 2014
id: 392
Title: From the Mixed-Up Files of Mrs. Basil E. Frankweiler
Genre: Fiction
Author: E.L. Konigsburg
Reviews:
id: 356 Rating: 3 Likes: 0 Date: August 24 2018
id: 396 Rating: 2 Likes: 558 Date: January 16 2011
id: 677 Rating: 5 Likes: 3 Date: August 13 2018
id: 914 Rating: 5 Likes: 0 Date: June 3 2019
id: 1654 Rating: 0 Likes: 0 Date: February 6 2019
id: 1665 Rating: 5 Likes: 0 Date: November 21 2017
id: 1866 Rating: 0 Likes: 0 Date: December 27 2018
id: 2173 Rating: 2 Likes: 0 Date: February 17 2012
id: 2285 Rating: 5 Likes: 0 Date: March 28 2013
id: 2901 Rating: 5 Likes: 0 Date: July 20 2017
id: 4481 Rating: 5 Likes: 2 Date: May 21 2016
id: 5827 Rating: 4 Likes: 0 Date: November 25 2019
id: 5975 Rating: 1 Likes: 0 Date: March 28 2018
id: 6899 Rating: 5 Likes: 1 Date: February 27 2018
id: 8914 Rating: 1 Likes: 0 Date: July 12 2013
id: 9248 Rating: 5 Likes: 0 Date: October 4 2018
id: 9999 Rating: 5 Likes: 0 Date: March 18 2015
id: 10317 Rating: 4 Likes: 0 Date: April 18 2014
id: 10521 Rating: 4 Likes: 0 Date: September 13 2019
id: 10822 Rating: 5 Likes: 0 Date: May 11 2016
id: 11689 Rating: 5 Likes: 1 Date: January 27 2017
id: 12274 Rating: 5 Likes: 0 Date: August 25 2015
id: 12379 Rating: 5 Likes: 0 Date: June 23 2015
id: 12431 Rating: 0 Likes: 1 Date: July 25 2019
id: 12626 Rating: 3 Likes: 0 Date: September 18 2019
id: 13444 Rating: 0 Likes: 0 Date: June 26 2013
id: 13813 Rating: 3 Likes: 0 Date: May 27 2018
id: 14083 Rating: 3 Likes: 0 Date: October 26 2014
id: 14142 Rating: 4 Likes: 0 Date: December 27 2014
id: 14199 Rating: 3 Likes: 4 Date: October 12 2015
id: 14427 Rating: 1 Likes: 2 Date: June 15 2018
id: 16431 Rating: 5 Likes: 0 Date: October 28 2014
id: 16509 Rating: 5 Likes: 0 Date: March 17 2016
id: 16740 Rating: 3 Likes: 0 Date: April 26 2016
id: 16829 Rating: 0 Likes: 0 Date: October 15 2017
id: 17340 Rating: 3 Likes: 0 Date: November 13 2017
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
id: 126
Title: Diva
Genre: Historical
Author: Jillian Larkin
Reviews:
id: 139 Rating: 3 Likes: 0 Date: May 2 2019
id: 352 Rating: 5 Likes: 69 Date: January 19 2019
id: 735 Rating: 0 Likes: 0 Date: September 26 2014
id: 986 Rating: 2 Likes: 0 Date: February 2 2010
id: 1590 Rating: 4 Likes: 0 Date: June 16 2014
id: 1922 Rating: 1 Likes: 0 Date: March 14 2014
id: 2169 Rating: 5 Likes: 0 Date: February 13 2019
id: 2589 Rating: 3 Likes: 0 Date: January 24 2017
id: 4273 Rating: 4 Likes: 15 Date: August 7 2015
id: 7605 Rating: 2 Likes: 0 Date: June 29 2019
id: 7689 Rating: 0 Likes: 0 Date: July 3 2018
id: 8101 Rating: 2 Likes: 0 Date: September 18 2019
id: 8195 Rating: 1 Likes: 0 Date: August 11 2015
id: 8542 Rating: 4 Likes: 0 Date: April 21 2019
id: 8727 Rating: 5 Likes: 0 Date: April 4 2012
id: 9078 Rating: 1 Likes: 0 Date: September 10 2010
id: 9288 Rating: 2 Likes: 0 Date: May 26 2019
id: 10364 Rating: 4 Likes: 0 Date: January 27 2014
id: 10655 Rating: 1 Likes: 26 Date: February 3 2015
id: 10780 Rating: 5 Likes: 69 Date: March 25 2018
id: 10864 Rating: 2 Likes: 30 Date: June 19 2014
id: 11013 Rating: 3 Likes: 39 Date: August 21 2019
id: 11044 Rating: 4 Likes: 2 Date: May 17 2019
id: 11592 Rating: 5 Likes: 1 Date: December 17 2017
id: 11745 Rating: 3 Likes: 0 Date: March 14 2013
id: 11830 Rating: 1 Likes: 0 Date: May 3 2010
id: 11840 Rating: 5 Likes: 0 Date: April 27 2014
id: 12010 Rating: 0 Likes: 12 Date: March 16 2014
id: 12491 Rating: 3 Likes: 0 Date: May 3 2018
id: 14089 Rating: 5 Likes: 1 Date: July 4 2011
id: 14123 Rating: 3 Likes: 2 Date: November 8 2018
id: 14318 Rating: 3 Likes: 1 Date: April 12 2018
id: 14999 Rating: 0 Likes: 0 Date: April 8 2011
id: 15041 Rating: 1 Likes: 0 Date: August 9 2009
id: 15364 Rating: 5 Likes: 0 Date: December 29 2018
id: 15486 Rating: 2 Likes: 0 Date: July 17 2017
id: 15570 Rating: 4 Likes: 0 Date: June 22 2019
id: 15906 Rating: 0 Likes: 5 Date: October 7 2017
id: 16287 Rating: 5 Likes: 0 Date: December 17 2019
id: 16779 Rating: 5 Likes: 0 Date: May 7 2014
id: 16961 Rating: 0 Likes: 0 Date: October 21 2018
id: 17016 Rating: 5 Likes: 0 Date: July 25 2018
0.005969
id: 177
Title: Disasterology 101
Genre: Romance
Author: Taylor V. Donovan
Reviews:
id: 973 Rating: 1 Likes: 0 Date: September 10 2019
id: 1143 Rating: 2 Likes: 2 Date: March 27 2015
id: 1343 Rating: 0 Likes: 0 Date: July 6 2013
id: 1444 Rating: 2 Likes: 0 Date: February 6 2016
id: 1753 Rating: 3 Likes: 0 Date: June 2 2016
id: 2235 Rating: 5 Likes: 1 Date: November 27 2015
id: 2375 Rating: 2 Likes: 0 Date: May 1 2019
id: 3060 Rating: 5 Likes: 0 Date: September 3 2018
id: 3925 Rating: 5 Likes: 0 Date: June 15 2012
id: 4134 Rating: 5 Likes: 0 Date: August 17 2017
id: 5279 Rating: 5 Likes: 0 Date: January 14 2016
id: 5280 Rating: 2 Likes: 0 Date: May 10 2016
id: 6593 Rating: 3 Likes: 9 Date: February 13 2017
id: 6981 Rating: 2 Likes: 2 Date: June 29 2017
id: 8636 Rating: 4 Likes: 0 Date: February 13 2019
id: 8688 Rating: 1 Likes: 0 Date: May 10 2012
id: 9550 Rating: 3 Likes: 0 Date: October 1 2018
id: 10543 Rating: 0 Likes: 0 Date: August 3 2010
id: 10658 Rating: 4 Likes: 5 Date: February 7 2015
id: 10661 Rating: 1 Likes: 5 Date: November 7 2019
id: 10964 Rating: 2 Likes: 90 Date: March 23 2016
id: 11483 Rating: 1 Likes: 11 Date: January 5 2019
id: 11667 Rating: 4 Likes: 0 Date: May 17 2019
id: 12054 Rating: 3 Likes: 3 Date: February 29 2011
id: 13276 Rating: 3 Likes: 0 Date: December 17 2017
id: 15316 Rating: 4 Likes: 0 Date: August 23 2015
id: 15368 Rating: 3 Likes: 0 Date: June 4 2019
id: 15515 Rating: 4 Likes: 0 Date: February 14 2019
id: 16433 Rating: 0 Likes: 0 Date: October 28 2018
id: 16793 Rating: 5 Likes: 0 Date: January 24 2015
id: 16830 Rating: 3 Likes: 0 Date: February 13 2017
id: 31
Name: Athena Souri
Place of Birth: Switzerland
Member Since: December 24 2010
Favorite Genres: Romance
Favorite Authors: Carl Sagan, Colleen Clayton, Greer Noble
Number of Books in Read Shelf: 8
Number of Books in Want to Read Shelf: 15
Number of Books in Currently Reading Shelf: 6
Number of Likes: 2455
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
id: 236
Title: Slow Burn
Genre: Romance
Author: Sam B. Morgan
Reviews:
id: 22 Rating: 0 Likes: 38 Date: February 27 2016
id: 709 Rating: 4 Likes: 2 Date: September 27 2013
id: 816 Rating: 2 Likes: 0 Date: November 29 2019
id: 1315 Rating: 5 Likes: 0 Date: October 16 2015
id: 1626 Rating: 2 Likes: 0 Date: March 28 2019
id: 1698 Rating: 3 Likes: 0 Date: July 2 2016
id: 2240 Rating: 2 Likes: 0 Date: March 4 2013
id: 2338 Rating: 3 Likes: 0 Date: March 29 2019
id: 2651 Rating: 2 Likes: 0 Date: November 25 2018
id: 2702 Rating: 5 Likes: 0 Date: October 5 2018
id: 2861 Rating: 3 Likes: 0 Date: April 11 2014
id: 4305 Rating: 5 Likes: 20 Date: May 8 2015
id: 4625 Rating: 0 Likes: 1 Date: March 8 2011
id: 5276 Rating: 5 Likes: 0 Date: September 4 2019
id: 6661 Rating: 4 Likes: 2 Date: January 11 2019
id: 6698 Rating: 3 Likes: 2 Date: December 29 2010
id: 6754 Rating: 5 Likes: 5 Date: July 23 2014
id: 7207 Rating: 4 Likes: 13 Date: June 27 2018
id: 7241 Rating: 3 Likes: 12 Date: October 20 2011
id: 7335 Rating: 1 Likes: 1 Date: April 15 2012
id: 7395 Rating: 2 Likes: 0 Date: September 9 2019
id: 7785 Rating: 2 Likes: 0 Date: January 27 2019
id: 7998 Rating: 4 Likes: 0 Date: March 19 2016
id: 8669 Rating: 3 Likes: 0 Date: May 2 2014
id: 9719 Rating: 0 Likes: 0 Date: November 24 2019
id: 9874 Rating: 2 Likes: 0 Date: August 25 2014
id: 10004 Rating: 4 Likes: 0 Date: March 25 2019
id: 10171 Rating: 4 Likes: 2 Date: February 14 2019
id: 10644 Rating: 2 Likes: 0 Date: June 25 2019
id: 10714 Rating: 4 Likes: 20 Date: June 6 2019
id: 10841 Rating: 3 Likes: 0 Date: April 27 2017
id: 10909 Rating: 5 Likes: 0 Date: April 9 2013
id: 11757 Rating: 0 Likes: 0 Date: March 27 2018
id: 12179 Rating: 5 Likes: 0 Date: September 5 2015
id: 13019 Rating: 5 Likes: 1 Date: May 21 2012
id: 13321 Rating: 4 Likes: 0 Date: May 18 2016
id: 14914 Rating: 1 Likes: 0 Date: November 17 2018
id: 16115 Rating: 5 Likes: 0 Date: April 7 2016
id: 211
Title: The Fall of Five
Genre: Young Adult
Author: Pittacus Lore
Reviews:
id: 76 Rating: 5 Likes: 0 Date: August 19 2016
id: 490 Rating: 4 Likes: 0 Date: January 26 2018
id: 1006 Rating: 1 Likes: 3 Date: January 25 2017
id: 1861 Rating: 4 Likes: 0 Date: February 9 2018
id: 2686 Rating: 0 Likes: 0 Date: September 11 2016
id: 2727 Rating: 5 Likes: 0 Date: November 26 2015
id: 3688 Rating: 2 Likes: 0 Date: September 9 2017
id: 3734 Rating: 1 Likes: 0 Date: April 15 2012
id: 4304 Rating: 3 Likes: 26 Date: September 21 2011
id: 4478 Rating: 4 Likes: 1 Date: March 27 2019
id: 4593 Rating: 5 Likes: 0 Date: January 16 2015
id: 4677 Rating: 4 Likes: 1 Date: September 1 2016
id: 4857 Rating: 5 Likes: 4 Date: September 19 2017
id: 5397 Rating: 5 Likes: 0 Date: October 10 2018
id: 6087 Rating: 1 Likes: 0 Date: June 9 2011
id: 6101 Rating: 4 Likes: 0 Date: March 9 2019
id: 7230 Rating: 1 Likes: 10 Date: December 29 2017
id: 7660 Rating: 5 Likes: 3 Date: April 24 2018
id: 8166 Rating: 5 Likes: 0 Date: January 6 2015
id: 8699 Rating: 1 Likes: 0 Date: July 25 2019
id: 9124 Rating: 0 Likes: 0 Date: February 9 2016
id: 9144 Rating: 3 Likes: 0 Date: December 15 2017
id: 9730 Rating: 1 Likes: 0 Date: June 9 2011
id: 9892 Rating: 2 Likes: 0 Date: May 27 2017
id: 10053 Rating: 5 Likes: 0 Date: October 12 2018
id: 10413 Rating: 5 Likes: 0 Date: December 14 2018
id: 10697 Rating: 0 Likes: 22 Date: April 26 2017
id: 10906 Rating: 0 Likes: 0 Date: October 8 2014
id: 10993 Rating: 1 Likes: 0 Date: October 20 2018
id: 11111 Rating: 2 Likes: 3 Date: February 28 2016
id: 11330 Rating: 0 Likes: 26 Date: September 29 2014
id: 11761 Rating: 5 Likes: 0 Date: April 24 2015
id: 11958 Rating: 3 Likes: 0 Date: October 23 2014
id: 12401 Rating: 1 Likes: 0 Date: August 13 2015
id: 12622 Rating: 0 Likes: 0 Date: November 12 2019
id: 13928 Rating: 2 Likes: 0 Date: November 29 2019
id: 14234 Rating: 2 Likes: 2 Date: March 15 2019
id: 14242 Rating: 1 Likes: 1 Date: April 4 2019
id: 14568 Rating: 0 Likes: 0 Date: February 27 2018
id: 16016 Rating: 3 Likes: 0 Date: December 20 2015
id: 16289 Rating: 4 Likes: 0 Date: August 14 2015
id: 16570 Rating: 4 Likes: 0 Date: October 13 2018
id: 16628 Rating: 1 Likes: 0 Date: January 10 2016
id: 16765 Rating: 4 Likes: 0 Date: October 27 2018
id: 211
Title: The Fall of Five
Genre: Young Adult
Author: Pittacus Lore
Reviews:
id: 76 Rating: 5 Likes: 0 Date: August 19 2016
id: 490 Rating: 4 Likes: 0 Date: January 26 2018
id: 1006 Rating: 1 Likes: 3 Date: January 25 2017
id: 1861 Rating: 4 Likes: 0 Date: February 9 2018
id: 2686 Rating: 0 Likes: 0 Date: September 11 2016
id: 2727 Rating: 5 Likes: 0 Date: November 26 2015
id: 3688 Rating: 2 Likes: 0 Date: September 9 2017
id: 3734 Rating: 1 Likes: 0 Date: April 15 2012
id: 4304 Rating: 3 Likes: 26 Date: September 21 2011
id: 4478 Rating: 4 Likes: 1 Date: March 27 2019
id: 4593 Rating: 5 Likes: 0 Date: January 16 2015
id: 4677 Rating: 4 Likes: 1 Date: September 1 2016
id: 4857 Rating: 5 Likes: 4 Date: September 19 2017
id: 5397 Rating: 5 Likes: 0 Date: October 10 2018
id: 6087 Rating: 1 Likes: 0 Date: June 9 2011
id: 6101 Rating: 4 Likes: 0 Date: March 9 2019
id: 7230 Rating: 1 Likes: 10 Date: December 29 2017
id: 7660 Rating: 5 Likes: 3 Date: April 24 2018
id: 8166 Rating: 5 Likes: 0 Date: January 6 2015
id: 8699 Rating: 1 Likes: 0 Date: July 25 2019
id: 9124 Rating: 0 Likes: 0 Date: February 9 2016
id: 9144 Rating: 3 Likes: 0 Date: December 15 2017
id: 9730 Rating: 1 Likes: 0 Date: June 9 2011
id: 9892 Rating: 2 Likes: 0 Date: May 27 2017
id: 10053 Rating: 5 Likes: 0 Date: October 12 2018
id: 10413 Rating: 5 Likes: 0 Date: December 14 2018
id: 10697 Rating: 0 Likes: 22 Date: April 26 2017
id: 10906 Rating: 0 Likes: 0 Date: October 8 2014
id: 10993 Rating: 1 Likes: 0 Date: October 20 2018
id: 11111 Rating: 2 Likes: 3 Date: February 28 2016
id: 11330 Rating: 0 Likes: 26 Date: September 29 2014
id: 11761 Rating: 5 Likes: 0 Date: April 24 2015
id: 11958 Rating: 3 Likes: 0 Date: October 23 2014
id: 12401 Rating: 1 Likes: 0 Date: August 13 2015
id: 12622 Rating: 0 Likes: 0 Date: November 12 2019
id: 13928 Rating: 2 Likes: 0 Date: November 29 2019
id: 14234 Rating: 2 Likes: 2 Date: March 15 2019
id: 14242 Rating: 1 Likes: 1 Date: April 4 2019
id: 14568 Rating: 0 Likes: 0 Date: February 27 2018
id: 16016 Rating: 3 Likes: 0 Date: December 20 2015
id: 16289 Rating: 4 Likes: 0 Date: August 14 2015
id: 16570 Rating: 4 Likes: 0 Date: October 13 2018
id: 16628 Rating: 1 Likes: 0 Date: January 10 2016
id: 16765 Rating: 4 Likes: 0 Date: October 27 2018
id: 405
Title: Ellen Foster
Genre: Fiction
Author: Kaye Gibbons
Average Rating: 3.42
